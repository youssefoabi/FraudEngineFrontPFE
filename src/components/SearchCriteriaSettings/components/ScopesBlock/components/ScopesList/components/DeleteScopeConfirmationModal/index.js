import { bool, func, string } from 'prop-types';
import { isNilOrEmpty } from 'ramda-adjunct';
import { Typography } from '@material-ui/core';
import React from 'react';

import ConfirmationModal from '../../../../../../../ConfirmationModal';

export default function DeleteEnginesConfirmationModal({ name, confirm, open, toggle }) {
  if (isNilOrEmpty(name)) return null;

  return (
    <ConfirmationModal isOpen={open} title="Suppression de la gamme" toggle={toggle} confirm={confirm}>
      <Typography>{`Confirmez-vous la suppression de la gamme  ${name} ?`}</Typography>
    </ConfirmationModal>
  );
}

DeleteEnginesConfirmationModal.propTypes = {
  open: bool,
  toggle: func,
  name: string,
  confirm: func,
};

DeleteEnginesConfirmationModal.defaultProps = {
  open: false,
  toggle: Function.prototype,
  name: null,
  confirm: Function.prototype,
};
