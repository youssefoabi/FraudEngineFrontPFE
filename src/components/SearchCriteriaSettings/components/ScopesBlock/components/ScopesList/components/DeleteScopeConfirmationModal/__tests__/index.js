import { itRendersAllMutations } from '@cdiscount/jest-utils';

import DeleteScopeConfirmationModal from '../';

const mutations = [
  { name: 'with scopes', props: { name: 'scope', confirm: jest.fn(), open: true, toggle: jest.fn() } },
  { name: 'without props' },
];

describe(DeleteScopeConfirmationModal.name, () => {
  itRendersAllMutations(DeleteScopeConfirmationModal, mutations);
});
