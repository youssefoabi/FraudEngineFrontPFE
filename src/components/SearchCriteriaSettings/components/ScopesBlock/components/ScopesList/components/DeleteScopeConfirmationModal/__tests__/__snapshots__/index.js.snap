// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteEnginesConfirmationModal DeleteEnginesConfirmationModal defaultProp functions runs defaultProps.confirm() 1`] = `undefined`;

exports[`DeleteEnginesConfirmationModal DeleteEnginesConfirmationModal defaultProp functions runs defaultProps.toggle() 1`] = `undefined`;

exports[`DeleteEnginesConfirmationModal DeleteEnginesConfirmationModal renders correctly with scopes 1`] = `
<ConfirmationModal
  confirm={[MockFunction]}
  isOpen={true}
  title="Suppression de la gamme"
  toggle={[MockFunction]}
>
  <WithStyles(ForwardRef(Typography))>
    Confirmez-vous la suppression de la gamme  scope ?
  </WithStyles(ForwardRef(Typography))>
</ConfirmationModal>
`;

exports[`DeleteEnginesConfirmationModal DeleteEnginesConfirmationModal renders correctly without props 1`] = `""`;
